packages/shared-components/src/structures/RoomStatusBarHistoryVisible.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2024 New Vector Ltd.
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
+Please see LICENSE files in the repository root for full details.
+*/
+
+import React, { type ReactElement } from "react";
+import classNames from "classnames";
+import { Alert, Button } from "@vector-im/compound-web";
+
+import styles from "./RoomStatusBarHistoryVisible.module.css";
+import { _t } from "../utils/i18n";
+
+interface RoomStatusBarHistoryVisibleProps {
+    /**
+     * Called when the user presses the "dismiss" button.
+     */
+    onClose: () => void;
```

Comment:
```suggestion
    onClose: MouseEventHandler<HTMLButton>;
```

Add this import too: `import { type MouseEventHandler } from "react";`